src/ReasonReactRouter.re

```diff
@@ -145,11 +160,18 @@ let urlNotEqual = (a, b) => {
         listNotEqual(aRest, bRest);
       }
     };
-  };
   a.hash !== b.hash || a.search !== b.search || listNotEqual(a.path, b.path);
 };
 type watcherID = unit => unit;
-let url = () => {path: path(), hash: hash(), search: search()};
+let url = (~serverUrlString=?, ()) =>
+  switch (serverUrlString) {
```

Comment:
@rickyvetter I thinkered with this, but I can't find a way to do this without cheking `~serverUrlString` like this, perhaps there's some reason featured I am not aware of, I know about `Belt`, but since I didn't see it used anywhere I thought this was a design choice, so for now I've added a switch to check what `~serverUrlString`  is before returning a new `url`, please let me know if there's any way to improve this and I'll get back to working on it.